static/app/utils/url/safeURL.spec.tsx

```diff
@@ -0,0 +1,136 @@
+import {safeURL} from './safeURL';
+
+function makeURLMockWithCanParse(spy: jest.Mock): [typeof URL, jest.Mock] {
+  class URLMockWithCanParse extends URL {
+    constructor(...instanceArgs: any[]) {
+      // @ts-expect-error ???
+      super(...instanceArgs);
+      spy(...instanceArgs);
+    }
+
+    static canParse(): boolean {
+      throw new Error("Specify canParse's return value");
+    }
+  }
+
+  return [URLMockWithCanParse, spy];
+}
+
+function makeURLMockWithoutCanParse(spy: jest.Mock): [typeof URL, jest.Mock] {
+  class URLMockWithCanParse extends URL {
+    constructor(...instanceArgs: any[]) {
+      // @ts-expect-error ???
```

Comment:
if the expect-error because we've got `any[]` up there?

Reply:
iirc it's saying it has to be some tuple arg and I just didn't wanna do mental gymnastics :D I'll try ConstructorParameters

Reply:
either way it's low-pri in a test file imo

Reply:
I can actually remove this based on @anonrig's comment about double parsing.